src/mon/OSDMonitor.cc

```diff
@@ -796,14 +796,20 @@ class OSDUtilizationDumper : public CrushTreeDumper::Dumper<F> {
 
   bool get_osd_utilization(int id, int64_t* kb, int64_t* kb_used,
 			   int64_t* kb_avail) const {
+    // we always return true for OSDs as the specific OSDs can be out,
+    // and we want to continue processing other OSDs under this bucket.
     typedef ceph::unordered_map<int32_t,osd_stat_t> OsdStat;
     OsdStat::const_iterator p = pgm->osd_stat.find(id);
-    if (p == pgm->osd_stat.end())
-      return false;
+    if (p == pgm->osd_stat.end()) {
+      *kb = 0;
+      *kb_used = 0;
+      *kb_avail = 0;
+      return true;
+    }
     *kb = p->second.kb;
     *kb_used = p->second.kb_used;
     *kb_avail = p->second.kb_avail;
-    return *kb > 0;
+    return true;
```

Comment:
I thought the out OSD should be caught by the condition https://github.com/ceph/ceph/pull/10308/files#diff-0a5db46a44ae9900e226289a810f10e8R803 at first, but it turned out the out OSD's statistics is zeroed. So this change is necessary.